tests/unit/parallel/executors/shared_parallel_executor.cpp

```diff
@@ -32,6 +33,17 @@ struct shared_parallel_executor
     {
         return hpx::async(std::forward<F>(f));
     }
+
+    template <typename F>
+    typename std::remove_reference<
+        typename hpx::util::result_of<
+            typename hpx::util::decay<F>::type()
+        >::type
+    >::type
```

Comment:
On 02/18/2016 01:36 PM, Hartmut Kaiser wrote:
> In tests/unit/parallel/executors/shared_parallel_executor.cpp
> <https://github.com/STEllAR-GROUP/hpx/pull/1992#discussion_r53307776>:
> 
>> @@ -32,6 +33,17 @@ struct shared_parallel_executor
>>      {
>>          return hpx::async(std::forward<F>(f));
>>      }
>> +
>> +    template <typename F>
>> +    typename std::remove_reference<
>> +        typename hpx::util::result_of<
>> +            typename hpx::util::decay<F>::type()
>> +        >::type
>> +    >::type
> 
> I tried solving it in the trait, but was not able to do it there. If you
> have an idea how this could be formulated, I'd appreciate any ideas.

I am still trying to wrap my around why this is needed in the first place.

> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/STEllAR-GROUP/hpx/pull/1992/files#r53307776>.
> 


-- 
Thomas Heller
Friedrich-Alexander-Universität Erlangen-Nürnberg
Department Informatik - Lehrstuhl Rechnerarchitektur
Martensstr. 3
91058 Erlangen
Tel.: 09131/85-27018
Fax:  09131/85-27912
Email: thomas.heller@cs.fau.de